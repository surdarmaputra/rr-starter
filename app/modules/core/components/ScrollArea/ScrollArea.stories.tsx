import type { Meta, StoryObj } from '@storybook/react';

import { ScrollArea, type ScrollAreaProps } from './ScrollArea';

type Story = StoryObj<typeof ScrollArea>;

const meta: Meta = {
  component: ScrollArea,
  title: 'Core/Components/ScrollArea',
};

export default meta;

export const Default: Story = {
  render: (args: ScrollAreaProps) => (
    <ScrollArea className="h-[200px] w-[350px] rounded-md border p-4" {...args}>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris eu
      vestibulum massa. Morbi euismod, nunc ac dignissim euismod, velit justo
      faucibus nisl, nec malesuada turpis nunc a metus. Sed auctor, magna quis
      blandit venenatis, metus nisl tincidunt lorem, sit amet rutrum libero
      lorem sit amet magna. Nullam ac erat sed libero ornare dictum ut et dolor.
      Pellentesque auctor, metus eu tincidunt convallis, massa erat vulputate
      sapien, ut malesuada nisl magna sit amet sapien. Nullam ac erat sed libero
      ornare dictum ut et dolor. Sed euismod, nunc ac dignissim euismod, velit
      justo faucibus nisl, nec malesuada turpis nunc a metus. Sed auctor, magna
      quis blandit venenatis, metus nisl tincidunt lorem, sit amet rutrum libero
      lorem sit amet magna. Nullam ac erat sed libero ornare dictum ut et dolor.
      Pellentesque auctor, metus eu tincidunt convallis, massa erat vulputate
      sapien, ut malesuada nisl magna sit amet sapien. Nullam ac erat sed libero
      ornare dictum ut et dolor.
    </ScrollArea>
  ),
};
