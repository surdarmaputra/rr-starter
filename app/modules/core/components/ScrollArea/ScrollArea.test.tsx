import { render, screen } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import { ScrollArea } from './ScrollArea';

describe('ScrollArea', () => {
  function renderComponent() {
    return render(
      <ScrollArea data-testid="ScrollArea-testid">Content</ScrollArea>,
    );
  }

  test('renders with data-testid', async () => {
    renderComponent();
    await screen.findByTestId('ScrollArea-testid');
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
