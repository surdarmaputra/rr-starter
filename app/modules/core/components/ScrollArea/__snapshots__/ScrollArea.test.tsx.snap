// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScrollArea > renders snapshot 1`] = `
<div>
  <div
    class="relative overflow-hidden"
    data-testid="ScrollArea-testid"
    dir="ltr"
    style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
  >
    <style>
      [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
    </style>
    <div
      class="h-full w-full rounded-[inherit]"
      data-radix-scroll-area-viewport=""
      style="overflow-x: hidden; overflow-y: scroll;"
    >
      <div
        style="min-width: 100%; display: table;"
      >
        Content
      </div>
    </div>
  </div>
</div>
`;
